src/generation_tools.jl

```diff
@@ -43,14 +41,80 @@ julia> idx_to_dim([(x,y) for x in 1:3, y in 1:3])
  1  2  3
 ```
 """
-function idx_to_dim(idx_arr::AbstractArray{T, N}) where {T, N} 
-    out_arr = Array{typeof(idx_arr[1][1])}(undef, size(idx_arr)..., length(idx_arr[begin]))
+function idx_to_dim(idx_arr::AbstractArray{T, N}) where {TT, NT, T<:NTuple{NT, TT}, N} 
```

Comment:
How should this even run?

```
julia> idx_to_dim(rand(1,2,3,4))
ERROR: MethodError: no method matching idx_to_dim(::Array{Float64, 4})
The function `idx_to_dim` exists, but no method is defined for this combination of argument types.

Closest candidates are:
  idx_to_dim(::AbstractArray{T, N}) where {TT, NT, T<:NTuple{NT, TT}, N}
   @ Main REPL[3]:25

Stacktrace:
 [1] top-level scope
   @ REPL[10]:1

```

Reply:
Yes, but I think it's the wrong way. I never used those functions and it just indicates something is fundamentally wrong at a different position.